challenge_take_out_the_garbage/src/challenge_take_out_the_garbage/pick_up.py

```diff
@@ -0,0 +1,143 @@
+# ROS
+import rospy
+import smach
+
+# TU/e
+import robot_skills
+import robot_smach_states as states
+import robot_smach_states.util.designators as ds
+
+from ed_msgs.msg import EntityInfo
+
+from robot_skills.util.kdl_conversions import FrameStamped, kdl_frame_from_XYZRPY
+
+
+class GetTrashBin(smach.State):
+    """
+    Get the entity of the trashbin in FrameStamped and save it to get the orientation of the fixed point in world model
+    """
+    def __init__(self, robot, trashbin):
+        smach.State.__init__(self, outcomes=["succeeded", "failed"])
+        self._robot = robot
+        self._trashbin = trashbin
+
+    def execute(self, userdata=None):
+        e = self._trashbin.resolve()
+        # get original entity pose
+        print("{}".format(type(self._robot.ed.get_entity(id=e.id)._pose)))
+
+        frame_original = self._robot.ed.get_entity(id=e.id)._pose
+
+        # inspect and update entity
+        states.look_at_segmentation_area(self._robot,  self._robot.ed.get_entity(id=e.id), 'on_top_of')
+        self._robot.ed.update_kinect("{} {}".format('on_top_of', e.id))
+        frame_updated = self._robot.ed.get_entity(id=e.id)._pose
+
+        # update entity with original orientation
+        test = frame_updated
+        test.M = frame_original.M
+        new_frame = FrameStamped(test, "map")
+
+        # new_frame.header.stamp
+        print("{}".format(new_frame))
+        self._robot.ed.update_entity(self._robot, e.id, frame_stamped=new_frame)
+        if e:
+            return "succeeded"
+        else:
+            return "failed"
+
+class GrabTrash(smach.State):
+    """
+    Set the arm in the prepare grasp position so it can safely go to the grasp position
+    """
+    def __init__(self, robot, arm_designator):
+        smach.State.__init__(self, outcomes=['succeeded', 'failed'])
+
+        self._robot = robot
+        self._arm_designator = arm_designator
+
+    def execute(self, userdata=None):
+
+        arm = self._arm_designator.resolve()
+        if not arm:
+            rospy.logerr("Could not resolve arm")
+            return "failed"
+
+        # Torso up (non-blocking)
+        self._robot.torso.reset()
+
+        # Arm to position in a safe way
+        arm.send_joint_goal('handover')
+        arm.wait_for_motion_done()
+
+        # Force drive to get closer to bin
+        self._robot.base.force_drive(0.15, 0, 0, 0.5)
```

Comment:
Put it in the knowledge, because this is trash bin depended.

Reply:
Is it? I would think this depends on your in front of area, which you can place at the desired position that incorporates this force drive, right? Since it is only used for this challenge, and you already have to tune the ```in_front_of``` area, also making a trash bin dependent force drive seems unnecessary, you can easily just leave it like this.